File: sdks/java/src/main/java/dev/atap/sdk/model/ApprovalSubject.java
package dev.atap.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

import java.util.Map;

/**
 * The purpose and payload of an approval.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class ApprovalSubject {

    @JsonProperty("type")
    private String type;

    @JsonProperty("label")
    private String label;

    @JsonProperty("reversible")
    private boolean reversible;

    @JsonProperty("payload")
    private Map<String, Object> payload;

    public ApprovalSubject() {
    }

    public ApprovalSubject(String type, String label, boolean reversible, Map<String, Object> payload) {
        this.type = type;
        this.label = label;
        this.reversible = reversible;
        this.payload = payload;
    }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getLabel() { return label; }
    public void setLabel(String label) { this.label = label; }

    public boolean isReversible() { return reversible; }
    public void setReversible(boolean reversible) { this.reversible = reversible; }

    public Map<String, Object> getPayload() { return payload; }
    public void setPayload(Map<String, Object> payload) { this.payload = payload; }
}

File: sdks/php/src/Model/ApprovalSubject.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Model;

/**
 * The purpose and payload of an approval.
 */
final class ApprovalSubject
{
    /**
     * @param array<string, mixed>|null $payload
     */
    public function __construct(
        public readonly string $type,
        public readonly string $label,
        public readonly bool $reversible = false,
        public readonly ?array $payload = null,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            type: (string) ($data['type'] ?? ''),
            label: (string) ($data['label'] ?? ''),
            reversible: (bool) ($data['reversible'] ?? false),
            payload: $data['payload'] ?? null,
        );
    }

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        return [
            'type' => $this->type,
            'label' => $this->label,
            'payload' => $this->payload ?? (object) [],
        ];
    }
}
